pandas/core/generic.py

```diff
@@ -466,9 +466,77 @@ def _expand_axes(self, key):
 
         return new_axes
 
-    def set_axis(self, axis, labels):
-        """ public verson of axis assignment """
-        setattr(self, self._get_axis_name(axis), labels)
+    _shared_docs['set_axis'] = """Assign desired index to given axis
+
+        Parameters
+        ----------
+        axis : int or string, default 0
+        labels: list-like or Index
+            The values for the new index
+        inplace : boolean, default False
```

Comment:
default is None (currently), it is *set* to True if its None (and the warning happens)